pkg/slo-controller/noderesource/plugins/batchresource/plugin.go

```diff
@@ -113,36 +119,47 @@ func (p *Plugin) Calculate(strategy *configuration.ColocationStrategy, node *cor
 			continue
 		}
 
+		// mark listed pod metric
+		podKey := util.GetPodKey(pod)
+		podMetric, hasMetric := podMetricMap[podKey]
```

Comment:
define another map to mark whether the podMetric has been added

Reply:
no longer change the map, use the subtraction between all pods' used and shown pods's used